### ISSUE TITLE
the `$family` object does not match the true family

### ISSUE THREAD

**Issue by FBartos** [bayesplay/bayesplay#9]
Even though that the `$family` is not meant to be accessed by users, I would expect it to match the family specified in the prior.
(note that `h3_mod$family` returns `cauchy` and `h4_mod$family` and `h5_mod$family` return `normal`)
```
h0_mod <- prior(family = "normal",    mean = 0, sd = 1)
h1_mod <- prior(family = "student_t", mean = 0, sd = 1, df = 10)
h2_mod <- prior(family = "cauchy",    location = 0, scale = 1)
h3_mod <- prior(family = "beta",      alpha = 1, beta = 1)
h4_mod <- prior(family = "uniform",   min = 0, max = 1)
h5_mod <- prior(family = "point",     point = 0)

h0_mod$family
h1_mod$family
h2_mod$family
h3_mod$family
h4_mod$family
h5_mod$family
```

Similarly, the `@function_text` does not return the expected print statement for `h1_mod$family`.
```
h0_mod@function_text
h1_mod@function_text
h2_mod@function_text
h3_mod@function_text
h4_mod@function_text
h5_mod@function_text
```

I actually implemented my own print function for your objects (and stored the proper family as the `priorType` attribute when creating them, might come handy for you if you decide to to change it in the future):
```

  priorText <- switch(
    attr(prior, "priorType"),
    "normal"  = paste0("Normal(", prior@data$parameters[1, "mean"], ", ", prior@data$parameters[1, "sd"], ")"),
    "t"       = paste0("Student-t(", prior@data$parameters[1, "mean"], ", ", prior@data$parameters[1, "sd"], ", ", prior@data$parameters[1, "df"], ")"),
    "cauchy"  = paste0("Cauchy(", prior@data$parameters[1, "location"], ", ", prior@data$parameters[1, "scale"], ")"),
    "beta"    = paste0("Beta(", prior@data$parameters[1, "alpha"], ", ", prior@data$parameters[1, "beta"], ")"),
    "uniform" = paste0("Uniform(", prior@data$parameters[1, "min"], ", ", prior@data$parameters[1, "max"], ")"),
    "spike"   = paste0("Spike(", prior@data$parameters[1, "point"], ")")
  )

  # add truncation if not default
  if (attr(prior, "priorType") == "beta" && (prior@data$parameters[1, "range"] != 0 || prior@data$parameters[2, "range"] != 1))
    priorText <- paste0(priorText, "[", prior@data$parameters[1, "range"], ",", prior@data$parameters[2, "range"] ,"]")

  if (attr(prior, "priorType") %in% c("normal", "t", "cauchy") && (!is.infinite(prior@data$parameters[1, "range"]) || !is.infinite(prior@data$parameters[2, "range"])))
    priorText <- paste0(priorText, "[", prior@data$parameters[1, "range"], ",", prior@data$parameters[2, "range"] ,"]")
```

**Comment by ljcolling** [comment]
Ok awesome. Thanks for this.

**Comment by ljcolling** [comment]
Fixed with #13

**Comment by ljcolling** [comment]
fixed with #16
